### PR TITLE
Formatter keeps quotes in atom keys

### PR DESCRIPTION
**Pull request by sabiwara**
Close https://github.com/elixir-lang/elixir/issues/13103

### CODE REVIEW

File: lib/elixir/lib/code/formatter.ex [hunk 1]
```diff
@@ -521,14 +521,15 @@ defmodule Code.Formatter do
...
 
             {:__block__, _, [atom]} when is_atom(atom) ->
-              key =
+              iodata =
                 if Macro.classify_atom(atom) in [:identifier, :unquoted] do
-                  IO.iodata_to_binary([Atom.to_string(atom), ?:])
+                  [Atom.to_string(atom), ?:]
                 else
-                  IO.iodata_to_binary([?", Atom.to_string(atom), ?", ?:])
+                  [?", atom |> Atom.to_string() |> String.replace("\"", "\\\""), ?", ?:]
```
Comment (josevalim) [inline lib/elixir/lib/code/formatter.ex:528]: I think we should use `Macro.inspect_atom(:key, atom)`. This way we can even skip the `Macro.classify_atom` call.

Comment (sabiwara) [reply lib/elixir/lib/code/formatter.ex:528]: Oh that's a nice one, totally forgot about inspect_atom!

But it seems to be over-escaping backslashes for our usage:
Screenshot 2023-11-15 at 20 55 05

Will try to figure out if we can remove the double escape.

Comment (sabiwara) [reply lib/elixir/lib/code/formatter.ex:528]: Also noticed that this code [here](https://github.com/elixir-lang/elixir/blob/main/lib/elixir/lib/code/formatter.ex#L1583) is almost the same, there's probably a way to refactor it (maybe with `Macro.inspect_atom/2`)

Comment (josevalim) [reply lib/elixir/lib/code/formatter.ex:528]: Ah, it is because strings in the formatted already unescaped. My bad. Your PR is good to go!

**Pull request merged**